docs/docs/core/08-events.md

```diff
@@ -11,8 +11,8 @@ sidebar_position: 1
 
 There are a few events that are automatically emitted for all messages, directly from `baseapp`.
 
-* `message.action`: The name of the message type.
-* `message.sender`: The address of the message signer.
+* `message.action`: The name of the message type. Exactly one exists for each transaction message, in the first ABCI `Event` of each transaction message.
+* `message.sender`: The address of the message signer. Exactly one exists for each transaction message, in the first ABCI `Event` of each transaction message.
 * `message.module`: The name of the module that emitted the message.
 
 :::tip
```

Comment:
If you don't mind:

```suggestion
* `message.action`: The name of the message type.
* `message.sender`: The address of the message signer.
* `message.module`: The name of the module that emitted the message.

:::tip
`baseapp` emits exactly one `message` event before any other events emitted by the message, which contains two attributes at least, `action` and `sender`. The position of the event may change in the next major version.
```

Reply:
@zemyblue and @0Tech, thank you for your comments.

I agree with you, and tried to improve an unclear point in @0Tech's suggestion. How about the following?

```
## Default Events

There are a few events that are automatically emitted for all messages, directly from `baseapp`.

* `message.action`: The name of the message type.
* `message.sender`: The address of the message signer.
* `message.module`: The name of the module that emitted the message.

:::tip
`baseapp` emits exactly one `message` event for each message before any other events emitted by the message.
The event contains at least 2 attributes, exactly one `action` and exactly one`sender`.
The position of the event may change in the next major version.
:::

:::tip
The module name is assumed by `baseapp` to be the second element of the message route: `"cosmos.bank.v1beta1.MsgSend" -> "bank"`.
In case a module does not follow the standard message path, (e.g. IBC), it is advised to keep emitting the module name event.
`Baseapp` only emits that event if the module have not already done so.
:::
```

Reply:
How about adding more detain about event like bellow. 
```
The `message` event contains at least 2 attributes, exactly one `action` and exactly one `sender`.
```
or
```
The `message` event contains at least two properties: 'action' and 'sender'.
```

Reply:
Thank you for the suggestion. I like the first one. It gives more information than the latter one.

```
The `message` event contains at least 2 attributes, exactly one `action` and exactly one `sender`.
```

Reply:
Please update document.